arch/x86/include/asm/barrier.h

```diff
@@ -19,9 +21,27 @@
 #define rmb() alternative("lock; addl $0,0(%%esp)", "lfence", X86_FEATURE_XMM2)
 #define wmb() alternative("lock; addl $0,0(%%esp)", "sfence", X86_FEATURE_XMM)
 #else
+#ifndef CONFIG_KTSAN
 #define mb() 	asm volatile("mfence":::"memory")
 #define rmb()	asm volatile("lfence":::"memory")
 #define wmb()	asm volatile("sfence" ::: "memory")
+#else /* CONFIG_KTSAN */
+#define mb()								\
+({									\
+	asm volatile("mfence":::"memory");				\
```

Comment:
If we do the atomic operation itself inside of other atomic functions, we should do the memory barrier itself inside of ktsan_thread_fence as well. You can do mfence always regardless of the actual memory ordering, it will not affect performance.